massgov/mayflower
# DP-15628 Expand a hot spot of TOC link to its parent container

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description


- Expand the clickable area of TOC to parent container of each link in mobile and desktop to maintain the consistent behavior.
- Change the cursor on the parent container to be a pointer to give a clear visual indicator as a link.
- Fix clicking the "Table of Contents" label in the mobile fly out doesn't jump to the top of the page.  
- Fix the TOC title covering the first 2 items in the mobile fly out.
- Match the hotspot in table-of-contents-hierarchy.twig to the sticky toc.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-15628)

## Steps to Test


1.  Go to below sample pages:
      - http://localhost:3000/patterns/05-pages-howto/05-pages-howto.html
      - http://localhost:3000/patterns/05-pages-information-details/05-pages-information-details.html
      - http://localhost:3000/patterns/05-pages-binder-page-cover/05-pages-binder-page-cover.html

2. In the desktop display, click outside the link text where shows a line to indicate a link below the text link, and see the area functions like a link.
<img width="477" alt="Screen Shot 2020-05-27 at 11 17 57 AM" src="https://user-images.githubusercontent.com/9633303/83039578-5a1b1f00-a00c-11ea-8052-33b206d39378.png">
<img width="678" alt="Screen Shot 2020-05-27 at 11 15 54 AM" src="https://user-images.githubusercontent.com/9633303/83039617-62735a00-a00c-11ea-9fa4-377fb37c3987.png">
<img width="666" alt="Screen Shot 2020-05-27 at 12 12 24 PM" src="https://user-images.githubusercontent.com/9633303/83045382-7f5f5b80-a013-11ea-8542-9a8e704d4777.png">


3. In the mobile display, do the same testing as step 2.

4. For the how to page sample, check all toc links are displayed in the mobile fly out.

<img width="680" alt="Mass_Gov" src="https://user-images.githubusercontent.com/9633303/83044341-0b708380-a012-11ea-888e-8b764a442a13.png">

5. For the info detail sample, check clicking the "Table of Contents" label in the mobile fly out doesn't jump up to the top of the page.


## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
